src/main/java/org/mitre/synthea/export/BB2RIFExporter.java

```diff
@@ -2569,46 +2607,88 @@ private long exportHome(Person person, long startTime, long stopTime) throws IOE
       fieldValues.put(HHA.REV_CNTR_RATE_AMT,
-          String.format("%.2f", encounter.claim.getTotalClaimCost()
+          String.format("%.2f", servicePeriod.getTotalCost().getTotalClaimCost()
                   .divide(BigDecimal.valueOf(days), RoundingMode.HALF_EVEN)
                   .setScale(2, RoundingMode.HALF_EVEN)));
       fieldValues.put(HHA.REV_CNTR_PMT_AMT_AMT,
-          String.format("%.2f", encounter.claim.getCoveredCost()));
+          String.format("%.2f", servicePeriod.getTotalCost().getCoveredCost()));
       fieldValues.put(HHA.REV_CNTR_TOT_CHRG_AMT,
-          String.format("%.2f", encounter.claim.getTotalClaimCost()));
+          String.format("%.2f", servicePeriod.getTotalCost().getTotalClaimCost()));
       fieldValues.put(HHA.REV_CNTR_NCVRD_CHRG_AMT,
-          String.format("%.2f", encounter.claim.getPatientCost()));
+          String.format("%.2f", servicePeriod.getTotalCost().getPatientCost()));
 
+      // random from fields TSV, may be overriden below
+      String revCenter = fieldValues.get(HHA.REV_CNTR);
+
+      final String HHA_TOTAL_CHARGE_REV_CNTR = "0001"; // Total charge
+      final String HHA_MEDICATION_REV_CNTR = "0270"; // General medical/surgical supplies
+      final String HHA_MEDICATION_CODE = "T1502"; // Administration of medication
+      ConsolidatedClaimLines consolidatedClaimLines = new ConsolidatedClaimLines();
+      for (HealthRecord.Encounter encounter : servicePeriod.getEncounters()) {
+        for (ClaimEntry lineItem : encounter.claim.items) {
+          String hcpcsCode = null;
+          if (lineItem.entry instanceof HealthRecord.Procedure) {
+            for (HealthRecord.Code code : lineItem.entry.codes) {
+              if (hcpcsCodeMapper.canMap(code.code)) {
+                hcpcsCode = hcpcsCodeMapper.map(code.code, person, true);
+                if (hhaRevCntrMapper.canMap(code.code)) {
+                  revCenter = hhaRevCntrMapper.map(code.code, person);
+                }
+                break; // take the first mappable code for each procedure
+              }
+            }
+            if (hcpcsCode == null) {
+              revCenter = HHA_TOTAL_CHARGE_REV_CNTR;
+            }
+            consolidatedClaimLines.addClaimLine(hcpcsCode, revCenter, lineItem);
+          } else if (lineItem.entry instanceof HealthRecord.Medication) {
+            HealthRecord.Medication med = (HealthRecord.Medication) lineItem.entry;
+            if (med.administration) {
+              hcpcsCode = HHA_MEDICATION_CODE;
+              revCenter = HHA_MEDICATION_REV_CNTR;
+              consolidatedClaimLines.addClaimLine(hcpcsCode, revCenter, lineItem);
+            }
+          }
+        }
+      }
+
+      // Use the final encounter in the service period to set all of the remaining field values that
+      // are the same for all claim lines
+      HealthRecord.Encounter encounter = servicePeriod.getEncounters().get(
+              servicePeriod.getEncounters().size() - 1);
```

Comment:
I'm not totally in love with this, because it is making some assumptions that the providers, clinicians, primary diagnosis reason, etc -- are all the same for all of the Encounters in the servicePeriod.

It isn't a terrible assumption, and it should mostly work. We can leave it for now.

Reply:
Good point. We could address that in the ConsolidatedServicePeriods class by using provider, clinician etc to stratify service periods. I'll add a TODO.